tests/integration-tests/tests/storage/test_efs.py

```diff
@@ -145,39 +158,128 @@ def create_efs():
             cfn_stacks_factory.delete_stack(stack.name, region)
 
 
-def _write_file_into_efs(region, vpc_stack, efs_id, request, key_name, cfn_stacks_factory):
-    """Write file stack contains a mount target and a instance to write a empty file with random name into the efs."""
+def _add_mount_targets(subnet_ids, efs_ids, security_group, template):
+    subnet_response = boto3.client("ec2").describe_subnets(SubnetIds=subnet_ids)["Subnets"]
+    for efs_index, efs_id in enumerate(efs_ids):
+        availability_zones_with_mount_target = set()
+        for mount_target in boto3.client("efs").describe_mount_targets(FileSystemId=efs_id)["MountTargets"]:
+            availability_zones_with_mount_target.add(mount_target["AvailabilityZoneName"])
+        for subnet_index, subnet in enumerate(subnet_response):
+            if subnet["AvailabilityZone"] not in availability_zones_with_mount_target:
+                # One and only one mount target should be created in each availability zone.
+                depends_on_arg = {}
+                resources = list(template.resources.keys())
+                if len(resources) >= 10:
+                    # Create at most 10 resources in parallel.
+                    depends_on_arg = {"DependsOn": [resources[-10]]}
+                template.add_resource(
+                    MountTarget(
+                        f"MountTargetResourceEfs{efs_index}Subnet{subnet_index}",
+                        FileSystemId=efs_id,
+                        SubnetId=subnet["SubnetId"],
+                        SecurityGroups=[Ref(security_group)],
+                        **depends_on_arg,
+                    )
+                )
+                availability_zones_with_mount_target.add(subnet["AvailabilityZone"])
+
+
+@pytest.fixture(scope="class")
+def mount_target_stack_factory(cfn_stacks_factory, request, region, vpc_stack):
+    """
+    EFS mount target stack.
+
+    Create mount targets in all availability zones of vpc_stack
+    """
+    created_stacks = []
+
+    def create_mount_targets(efs_ids):
+        template = Template()
+        template.set_version("2010-09-09")
+        template.set_description("Mount targets stack")
+
+        # Create security group that all communication between the mount targets and instances in the VPC
```

Comment:
`# Create security group that allows all communication between the mount targets and instances in the VPC`